elasticsearch/templates/statefulset.yaml

```diff
@@ -18,6 +18,7 @@ spec:
   selector:
     matchLabels:
       app: "{{ template "uname" . }}"
+      release: {{ .Release.Name | quote }}
```

Comment:
It looks like changing `spec.selector.matchLabels` during statefulset upgrade is failing:

```
 StatefulSet "web" is invalid: spec: Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', and 'updateStrategy' are forbidden
```

[elastic+helm-charts+pull-request+integration-elasticsearch/145/ES_SUITE=upgrade,KUBERNETES_VERSION=1.12](https://devops-ci.elastic.co/job/elastic+helm-charts+pull-request+integration-elasticsearch/145/ES_SUITE=upgrade,KUBERNETES_VERSION=1.12,label=docker&&virtual/consoleFull)

Reply:
I'm glad that upgrade test is in there and caught that!

Reply:
seems related to https://github.com/helm/charts/issues/7680